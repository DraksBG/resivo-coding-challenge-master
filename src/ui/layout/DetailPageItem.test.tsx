import { render, screen } from '@testing-library/react';
import { DetailPageItem } from './DetailPageItem';

describe('DetailPageItem', () => {

  it('renders the label correctly', () => {
    render(
      <DetailPageItem label="test">
        <div>Hello World!</div>
      </DetailPageItem>,
    );
    const labelElement = screen.getByText(/test/i);
    expect(labelElement).toBeInTheDocument();
  });

  it('renders the child element correctly', () => {
    render(
      <DetailPageItem label="test">
        <div>Hello World!</div>
      </DetailPageItem>,
    );
    const childElement = screen.getByText(/Hello World!/i);
    expect(childElement).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { container } = render(
      <DetailPageItem label="test">
        <div>Hello World!</div>
      </DetailPageItem>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

});
