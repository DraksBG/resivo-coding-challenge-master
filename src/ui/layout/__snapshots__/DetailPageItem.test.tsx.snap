// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPageItem matches the snapshot 1`] = `
<div
  class="MuiBox-root css-0"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    <strong>
      test
    </strong>
  </p>
  <div>
    Hello World!
  </div>
</div>
`;
